src/datajudge/constraints/base.py

```diff
@@ -241,6 +256,12 @@ def test(self, engine: sa.engine.Engine) -> TestResult:
             target_queries,
         )
 
+    def apply_output_formatting_no_counts(self, values: Collection) -> Collection:
```

Comment:
```suggestion
    def apply_output_formatting(self, values: Collection) -> Collection:
```
Not quite sure what the `no_counts` is for here.

Reply:
This function does not support getting an additional counts variable to consider in its output formatting (relevant for the subset constraint - since that special case only occurs once (so far), the relevant code is directly integrated in it. But yeah, this makes the name unnecessarily long - updated.